src/gui/dialogs/addon/addon_auth.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#define GETTEXT_DOMAIN "wesnoth-lib"
+
+#include <functional>
+
+#include "gui/dialogs/addon/addon_auth.hpp"
+
+#include "gui/auxiliary/find_widget.hpp"
+#include "gui/widgets/password_box.hpp"
+#include "gui/widgets/window.hpp"
+
+namespace gui2::dialogs
+{
+
+REGISTER_DIALOG(addon_auth)
+
+addon_auth::addon_auth(config& cfg)
+: cfg_(cfg)
+{
+	register_text("password", true);
```

Comment:
I just noticed you don't actually set an initial value here. You shouldn't need this line at all.

Reply:
Fixed.